src/index.ts

```diff
@@ -1,3 +1,89 @@
-export default function helloWorld(): string {
-	return 'Hello world!';
-}
+// Imports
+import {
+	CSSValidator,
+	ValidateTextOptions,
+	ValidateTextResultBase,
+	ValidateTextResultWithWarnings,
+	W3CCSSValidatorResponse,
+} from './types';
+import * as https from 'https';
+
+// Validates CSS using W3C's public CSS validator service
+const cssValidator: CSSValidator = {
+	// Validates a string of CSS
+	async validateText(unvalidatedStr: string, options?: ValidateTextOptions): Promise<any> {
+		// Build URL for fetching
+		const url = `https://jigsaw.w3.org/css-validator/validator?text=${encodeURIComponent(unvalidatedStr)}&usermedium=${
+			options?.medium ?? 'all'
+		}&warning=${options?.warningLevel ?? 0}&output=application/json`;
```

Comment:
We could make this more readable by defining the URL parameters in an object first, and then serializing those as URL parameters. Here's how to do that: https://stackoverflow.com/a/23639793/1250940.